apps/dashboard/app/models/batch_connect/session_context.rb

```diff
@@ -72,6 +66,15 @@ def update_with_cache(cache)
       self.attributes = cache.select { |k,v| self[k.to_sym] && self[k.to_sym].cacheable?(app_specific_cache_enabled?)  }
    end 
 
+  def to_openstruct(addons: {})
+    context_attrs = Hash[*map { |a| [a.id.to_sym, a.value] }.flatten]
```

Comment:
What does this line do? The `Hash[*map]` has me scratching my head because I'm unsure what the `*map` is being called with.

Reply:
It's the splat operator.  So `map` is going to return an array, but what we want is not an array (which is 1 thing), but _all the elements in the array_ (many things) (see the 2nd example here).

```
irb(main):008:0> Hash[*['a', 'b']]
=> {"a"=>"b"}
irb(main):009:0> Hash['a', 'b']
=> {"a"=>"b"}
```

You may see it a lot against hashes like so. So instead of passing in 1 single object (the hash), you're deconstructing the has so what gets passed is many key value pairs (named arguments) instead of 1 thing.
```ruby
foo = some_function_call(**my_args_hash)
```

https://thoughtbot.com/blog/ruby-splat-operator